cmd/container-disk-v1alpha/main.go

```diff
@@ -0,0 +1,175 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"time"
+
+	"kubevirt.io/kubevirt/pkg/log"
+)
+
+const (
+	EnvCopyPath            = "COPY_PATH"
+	EnvImagePath           = "IMAGE_PATH"
+	DiskSourceFallbackPath = "/disk"
+	QEMUIMGPath            = "/usr/bin/qemu-img"
+	ReadinessProbeFile     = "/tmp/healthy"
+)
+
+type DiskInfo struct {
+	Format      string `json:"format"`
+	BackingFile string `json:"backing-filename"`
+}
+
+func main() {
+	var err error
+
+	logger := log.DefaultLogger()
+
+	copyPath := os.Getenv(EnvCopyPath)
+	if copyPath == "" {
+		logger.Errorf("Environment Variable %s is not defined.", EnvCopyPath)
+		os.Exit(1)
+	}
+
+	imagePath := os.Getenv(EnvImagePath)
+	imagePath, err = GetImage(imagePath)
+	if err != nil {
+		logger.Reason(err).Errorf("Image lookup failed.")
+		os.Exit(1)
+	}
+
+	imageType, err := VerifyImage(imagePath)
+	if err != nil {
+		logger.Reason(err).Error("Could not determine image type.")
+		os.Exit(1)
+	}
+
+	err = CopyImage(imagePath, copyPath, imageType)
+	if err != nil {
+		logger.Reason(err).Errorf("Failed to copy image to the destination.")
+		os.Exit(1)
+	}
+
+	f, err := os.Create(ReadinessProbeFile)
+	if err != nil {
+		logger.Reason(err).Errorf("Failed to mark myself as ready.")
+		os.Exit(1)
+	}
+	f.Close()
+
+	for {
+		time.Sleep(1 * time.Second)
+	}
+}
+
+func GetImage(imagePath string) (string, error) {
+	if imagePath != "" {
+		if _, err := os.Stat(imagePath); err != nil {
+			if os.IsNotExist(err) {
+				return "", fmt.Errorf("No image on path %s", imagePath)
+			} else {
+				return "", fmt.Errorf("Failed to check if an image exists at %s", imagePath)
+			}
+		}
+	} else {
+		files, err := ioutil.ReadDir(DiskSourceFallbackPath)
```

Comment:
hm, at least from what I saw, it does not return an error if the directory does not have any files, so you will need to check that it does not equal to 0

Reply:
Yep, the code does that too, but the error message was nonsense. Fixed.